include/powerloader/curl.hpp

```diff
@@ -13,6 +13,11 @@
 #include <nlohmann/json.hpp>
 #include <tl/expected.hpp>
 
+extern "C"
+{
+#include <curl/curl.h>
+}
```

Comment:
We should definitely find a way to avoid leaking the curl API in powerloader. Again this can be done later since it was already the case before this PR.

Reply:
I agree, I suspect mosf ot the content of this header (curl.hpp) could be moved into the `src/curl_internal.hpp` header I added. We could try that soon indeed.